jetty-io/src/main/java/org/eclipse/jetty/io/AbstractConnection.java

```diff
@@ -311,24 +323,40 @@ public String toConnectionString()
             this);
     }
 
-    private class ReadCallback implements Callback
+    private class ReadCallback implements Callback, Invocable
     {
         @Override
         public void succeeded()
         {
+            if (isUseVirtualThreadToInvokeRootHandler())
+            {
+                if (VirtualThreads.startVirtualThread(AbstractConnection.this::onFillable))
+                    return;
+            }
```

Comment:
Instead of one `ReadCallback` class that has several methods that are essentially "if(virtual){startVirtual(task)}else{task}", why not just have 2 different `ReadCallback` implementations and pick one at construction time?